tests/lenchk

```diff
@@ -95,86 +97,83 @@ while [ "$1" ]; do
 
-# Thank you, Chubin. Feature apparently added in version 3.0 alpha, so should
-# be OK, but if anyone is having issues, just `cd` to the 'tests' directory.
-[ ${BASH_VERSINFO[0]:-0} -eq 3 ] || cd "${BASH_SOURCE[0]%/*}" &> /dev/null
+# Thank you, Chubin. Being in the 'tests' directory is expected. This handles
+# the edge case in which LenChk is instead sourced.
+cd "${BASH_SOURCE[0]%/*}" &> /dev/null
 
 # Confirm we are in the right place.
-git rev-parse --is-inside-work-tree 1> /dev/null 2>&1 ||
+git rev-parse --is-inside-work-tree &> /dev/null ||
 	Err 0 'Not inside a Git repository.'
 
 case $PWD in
 	*/cheat.sheets/tests)
 		;;
-	'')
-		Err 1 'Unable to determine the CWD.' ;;
 	*)
 		Err 1 "Not within the 'cheat.sheets/tests' directory." ;;
 esac
 
 Dirs=(../sheets/*)
-[ "$NoSubDirs" == 'True' ] || Dirs+=(../sheets/*/*)
+[[ $NoSubDirs == True ]] || Dirs+=(../sheets/*/*)
 
 # Add user's own directories to check, if they exist.
 for ArgDir in "$@"; {
-	if [ -d "$ArgDir" ]; then
-		Dirs+=($ArgDir/*)
+	if [[ -d $ArgDir ]]; then
+		Dirs+=("$ArgDir"/*)
 	else
 		Err 1 "Directory '$ArgDir' not found."
 	fi
 }
 
 # If the whitelist file exists, use it, unless whitelisting is disabled.
 # Keeping this test outside of the loop to avoid unnecessary processing.
-if [ "$NoWhiteList" != 'True' ] && [ -f "$WLFile" ]; then
+if [[ $NoWhiteList != True && -f $WLFile ]]; then
 	# Read the whitelist file, line-by-line, generating an array thereof.
-	Whitelisted=()
 	while read; do
 		Whitelisted+=("../${REPLY#cheat.sheets/}")
 	done < "$WLFile"
 fi
 
-# Using tput(1) for portability.
-if type -fP tput &> /dev/null; then
+# Prefer tput(1) for portability.
+if type -P tput &> /dev/null; then
 	C_Ellipses=`tput dim; tput setaf 7`
 	C_LineNums=`tput bold; tput setaf 2`
 	C_FileNames=`tput bold; tput setaf 1`
 	C_Reset=`tput sgr0`
 else
-	Err 1 'Unable to colorize output -- tput(1) not found.'
+	C_Ellipses='\e[2;37'
+	C_LineNums='\e[1;92m'
+	C_FileNames='\e[1;91m'
+	C_Reset='\e[0m'
 fi
 
-Main(){
+Main() {
 	for File in "${Dirs[@]}"; {
-		[ -f "$File" ] || continue
+		[[ -f $File ]] || continue
 
 		# Per chubin's desire to have an "in-bound flag"; see #134.
-		if [ "$NoLenChkLine" != 'True' ]; then
+		if [[ $NoLenChkLine != True ]]; then
 			SkipFile='False'
-
-			readarray -t Buffer < "$File"
-			for BufferLine in "${Buffer[@]}"; {
+			while read; do
 				if [[ $BufferLine == '# cheat.sh: '* ]]; then
 					CheatLine=${BufferLine#'# cheat.sh: '}
 					IFS=',' read -a Fields <<< "$CheatLine"
 					for Field in "${Fields[@]}"; {
-						[ "$Field" == "$Progrm=disable" ] && SkipFile='True'
+						[[ $Field == $Progrm=disable ]] && SkipFile='True'
 					}
 				fi
-			}
+			done < "$File"
 
-			[ "$SkipFile" == 'True' ] && continue
+			[[ $SkipFile == True ]] && continue
 		fi
 
 		# If the current file matches one which is whitelisted, skip it.
 		for CurWL in "${Whitelisted[@]}"; {
-			[ "$File" == "$CurWL" ] && continue 2
+			[[ $File == $CurWL ]] && continue 2
```

Comment:
And here around `$CurWL` too (but not around `$File` !)

Reply:
I will do the same here for the same reason as mentioned above.